pymc/model_graph.py

```diff
@@ -41,6 +43,118 @@ def fast_eval(var):
     return function([], var, mode="FAST_COMPILE")()
 
 
+class NodeType(str, Enum):
+    """Enum for the types of nodes in the graph."""
+
+    POTENTIAL = "Potential"
+    BASIC_RV = "Basic Random Variable"
+    OBSERVED_RV = "Observed Random Variable"
+    DETERMINISTIC = "Deterministic"
+    DATA = "Data"
+
+
+GraphvizNodeKwargs = dict[str, Any]
+NodeFormatter = Callable[[TensorVariable], GraphvizNodeKwargs]
+
+
+def default_potential(var: TensorVariable) -> GraphvizNodeKwargs:
+    """Default data for the node in the graph."""
+    return {
+        "shape": "octagon",
+        "style": "filled",
+        "label": f"{var.name}\n~\nPotential",
+    }
+
+
+def random_variable_symbol(var: TensorVariable) -> str:
+    """Get the name of the random variable."""
+    symbol = var.owner.op.__class__.__name__
+
+    if symbol.endswith("RV"):
+        symbol = symbol[:-2]
+
+    return symbol
+
+
+def default_basic_rv(var: TensorVariable) -> GraphvizNodeKwargs:
+    """Default data for the node in the graph."""
+    symbol = random_variable_symbol(var)
+
+    return {
+        "shape": "ellipse",
+        "style": None,
+        "label": f"{var.name}\n~\n{symbol}",
+    }
+
+
+def default_observed_rv(var: TensorVariable) -> GraphvizNodeKwargs:
+    """Default data for the node in the graph."""
+    symbol = random_variable_symbol(var)
+
+    return {
+        "shape": "ellipse",
+        "style": "filled",
+        "label": f"{var.name}\n~\n{symbol}",
+    }
+
+
+def default_deterministic(var: TensorVariable) -> GraphvizNodeKwargs:
+    """Default data for the node in the graph."""
+    return {
+        "shape": "box",
+        "style": None,
+        "label": f"{var.name}\n~\nDeterministic",
+    }
+
+
+def default_data(var: TensorVariable) -> GraphvizNodeKwargs:
+    """Default data for the node in the graph."""
+    return {
+        "shape": "box",
+        "style": "rounded, filled",
+        "label": f"{var.name}\n~\nData",
+    }
+
+
+def get_node_type(var_name: VarName, model) -> NodeType:
+    v = model[var_name]
+
+    if v in model.potentials:
```

Comment:
Small optimization, most models don't have potentials, and most have many free_RVs, and deterministics, so I would suggest changing the order of this to be something like `if deterministics elif free rvs elif observed rvs elif data elif potentials` instead